src/converse-service-administration.js

```diff
@@ -0,0 +1,452 @@
+// Converse.js (A browser based XMPP chat client)
+// https://conversejs.org
+//
+// Copyright (c) 2012-2017, Jan-Carel Brand <jc@opkode.com>
+// Licensed under the Mozilla Public License (MPLv2)
+//
+
+import "converse-chatview";
+import "converse-profile";
+import "converse-rosterview";
+import _FormData from "formdata-polyfill";
+import converse from "@converse/headless/converse-core";
+import tpl_service_administration_chat from "templates/service_administration_chat.html";
+import tpl_service_administration_option from "templates/service_administration_option.html";
+import tpl_service_administration_selector from "templates/service_administration_selector.html";
+
+const { $msg, $iq, Backbone, Promise, _ } = converse.env;
+const u = converse.env.utils;
+
+converse.plugins.add('converse-service-administration', {
+
+    /* Plugin dependencies are other plugins which might be
+     * overridden or relied upon, and therefore need to be loaded before
+     * this plugin.
+     *
+     * If the setting "strict_plugin_dependencies" is set to true,
+     * an error will be raised if the plugin is not found. By default it's
+     * false, which means these plugins are only loaded opportunistically.
+     *
+     * NB: These plugins need to have already been loaded via require.js.
+     */
+    dependencies: ["converse-chatboxes", "converse-rosterview", "converse-chatview"],
+
+    overrides: {},
+
+    initialize () {
+        const { _converse } = this,
+            { __ } = _converse;
+
+        _converse.api.listen.on('rosterViewInitialized', () => {
+            // check if service-administration is active. If not do not start
+            if (_converse.connection.connected) {
+                _converse.serviceAdministrationDiscovery = new _converse.ServiceAdministrationDiscovery({
+                    'controlbox_el': _converse.chatboxviews.get('controlbox').el.querySelector('.controlbox-panes') 
+                });
+                _converse.serviceAdministrationDiscovery.canUserSendBroadcastMessage();
+            }    
+        });
+
+        _converse.ServiceAdministrationDiscovery = Backbone.Model.extend({
+
+            // check if the user is allowed to send Broadcast-/MOTD-Messages
+            canUserSendBroadcastMessage () {
```

Comment:
This is not a good way to check for support.

From the XEP:
_In order to interact with a particular component attached to a server, an administrator needs to first discover that component and the commands it support, then send the appropriate command to the component itself._


Converse has general support for service discovery, so you can use the [_converse.api.disco](https://conversejs.org/docs/html/api/-_converse.api.disco.html) API.

For example:

`const supported = await _converse.api.disco.supports(Strophe.NS.COMMANDS, _converse.bare_jid);`.

You'll need to define `Strophe.NS.COMMANDS` in converse-core.js via `Strophe.addNamespace` (there are lots of examples). The value should be `http://jabber.org/protocol/commands`.

Once you know whether ad-hoc commands are supported, you can get a list of supported commands via:

`const commands = await _converse.api.disco.items(_converse.bare_jid, Strophe.NS.COMMANDS);`

You'll then know exactly which commands are supported by the server.